src/property-filter/controller.ts

```diff
@@ -127,21 +120,13 @@ export const parseText = (
   };
 };
 
-export const getPropertyOptions = (
-  filteringProperty: InternalFilteringProperty,
-  filteringOptions: readonly InternalFilteringOption[]
-) => {
-  return filteringOptions.filter(option => option.propertyKey === filteringProperty.propertyKey);
-};
-
 interface OptionGroup<T> {
   label: string;
   options: T[];
 }
 
 export const getAllValueSuggestions = (
-  filteringOptions: readonly InternalFilteringOption[],
-  filteringProperties: readonly InternalFilteringProperty[],
+  filteringSettings: FilteringSettings,
```

Comment:
This "knows it all" `filteringSettings` objects violates system design principles. Each component should receive only the minimum information it needs to operate. 

This "god object" breaks encapsulation between property filter sub components

Reply:
In the component the filtering properties and filtering options are used together more often than not. If you think it makes sense I can ensure the functions only receive the data they actually consume e.g. if only options are used - then only the options will be passed, not the entire settings object.

On the brighter side this settings structure helps in the following ways:
1. Encapsulates the relations between properties and options (and is going to be extended to also include definitions);
2. Performs all necessary mappings in one place so that in the downstream functions there is no longer a need to find properties by key or options by property key.

Reply:
Your "bright side" points is the book definition of the "god object" anti-pattern. Try to simplify individual components, and not create mega-controller (aka `filteringSettings`) where it is not needed.

Reply:
Also see another comment below. Found another red flag in this implementation: https://github.com/cloudscape-design/components/pull/1611#discussion_r1348626190